Download the latest otelcol for CI tests.

jimf5 opened the pull request:
### Proposed changes

Download the latest otelcol for CI tests.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-otel/blob/main/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes

p-pautov commented on .github/workflows/ubuntu.yml:
```diff
@@ -35,10 +35,13 @@ jobs:
-          curl -sLo - https://github.com/\
-          open-telemetry/opentelemetry-collector-releases/releases/download/\
-          v0.115.1/otelcol_0.115.1_linux_amd64.tar.gz | tar -xzv
+          curl -sLo - ${URL}download/otelcol_\
+          $(curl -sD - $URL | grep location: | cut -d v -f 2 | sed -e 's/\r//')\
+          _linux_amd64.tar.gz | tar -xzv
```
Using Github's API and jq would result in more readable and reliable code:
```suggestion
          LATEST=open-telemetry/opentelemetry-collector-releases/releases/latest
          TAG=$(curl -s https://api.github.com/repos/${LATEST} |
                jq -r .tag_name)
          curl -sLo - https://github.com/${LATEST}/download/\
          otelcol_${TAG:1}_linux_amd64.tar.gz | tar -xzv
```

p-pautov commented on .github/workflows/ubuntu.yml:
```diff
@@ -35,10 +35,13 @@ jobs:
                 -DNGX_OTEL_DEV=ON ..
           make -j $(nproc)
       - name: Download otelcol
+        env:
```
I don't think this section should be used for script's local variables.

jimf5 replied:
Yes, It's not needed anymore.  Removed.

p-pautov reviewed (commented):
The commit message needs more context, say:
> Download the latest otelcol for CI tests.

jimf5 commented:
> The commit message needs more context, say:
> 
> > Download the latest otelcol for CI tests.

Squashed all commits and updated the commit message.
